### PR TITLE
feat: Created confirmRegistration, rejectRegistration GraphQL-endpoints

### PR DESCRIPTION
**Pull request by Hyper3x**
#7848 

Created two new GraphQL endpoints confirmRegistration and rejectRegistration. Amended the confirm registration endpoint to be /records/{id}/confirm which was /confirm/registration.

### CODE REVIEW

File: packages/workflow/src/config/routes.ts [hunk 1]
```diff
@@ -59,9 +59,19 @@ export const getRoutes = () => {
...
+    // {
+    //   method: 'POST',
+    //   path: '/confirm/registration',
+    //   handler: markEventAsRegisteredCallbackHandler,
+    //   config: {
+    //     tags: ['api'],
+    //     description:
+    //       'Register event based on tracking id and registration number.'
+    //   }
+    // },
```
Comment (naftis) [inline packages/workflow/src/config/routes.ts:71]:
```suggestion

```

File: packages/gateway/src/features/registration/root-resolvers.ts [hunk 2]
```diff
@@ -636,20 +639,79 @@ export const resolvers: GQLResolver = {
...
       return taskEntry.resource.id
     },
-    async confirmRegistration(_, { id }, { headers: authHeader }) {
+    async confirmRegistration(_, { id, details }, { headers: authHeader }) {     
+      const token = authHeader.Authorization.replace("Bearer ", "").trim();
+      const recordSpecificToken = await tokenExchangeHandler(
+        token,
+        authHeader,
+        id
+      )
```
Comment (naftis) [inline packages/gateway/src/features/registration/root-resolvers.ts:648]: The confirmRegistration & rejectRegistration will be called with the record specific token - the token will already be in the headers. You don't need to fetch it anymore here

File: packages/gateway/src/features/registration/root-resolvers.ts [hunk 3]
```diff
@@ -636,20 +639,79 @@ export const resolvers: GQLResolver = {
...
+    ) {
+
+      const token = authHeader.Authorization.replace("Bearer ", "").trim();
+      const recordSpecificToken = await tokenExchangeHandler(
+        token,
+        authHeader,
+        id
+      )
+
+      authHeader.Authorization = `Bearer ${recordSpecificToken.access_token}`
```
Comment (naftis) [inline packages/gateway/src/features/registration/root-resolvers.ts:690]: Same here as in confirmRegistration

File: packages/gateway/src/features/registration/root-resolvers.ts [hunk 4]
```diff
@@ -636,20 +639,79 @@ export const resolvers: GQLResolver = {
...
+
+      if (!inScope(authHeader, ['record.reject-registration'])) {
+        throw new Error('User does not have a Reject Registration" scope');
       }
 
       if (!hasRecordAccess(authHeader, id)) {
-        throw new Error('User does not have access to the record')
+        throw new Error('User does not have access to the record');
```
Comment (naftis) [inline packages/gateway/src/features/registration/root-resolvers.ts:697]: Please check that you are using Prettier correctly - our config doesn't use semi-colons.

File: packages/migration/src/migrations/openhim/20220826062206-channels-v1.0.1-to-v1.1.0.ts [hunk 5]
```diff
@@ -52,7 +52,7 @@ const newChannel: Channel = {
   rewriteUrlsConfig: [],
   name: 'Confirm Registration',
   description: 'Confirm registration & assign a BRN',
-  urlPattern: '^/confirm/registration$',
+  urlPattern: '^/records/{id}/confirm$',
   priority: 1,
```
[the remark this answers is not in the document]
Comment (makelicious) [reply packages/migration/src/migrations/openhim/20220826062206-channels-v1.0.1-to-v1.1.0.ts:56]: Would this break down if you migrate from a version which has not run this? If it breaks, Is it a practical suggestion (there are no countries which use such version) or principled (countries should not be able to migrate from 1.2 to 1.7 directly)

File: packages/gateway/src/features/registration/root-resolvers.ts (hunk 3 above)
Comment (Hyper3x) [reply packages/gateway/src/features/registration/root-resolvers.ts:690]: Does this mean i have to remove this part? if i do so when i try to test it I am getting the user does not have the scope error message.

Comment (naftis) [reply packages/gateway/src/features/registration/root-resolvers.ts:690]: https://github.com/opencrvs/opencrvs-core/pull/7882 we finally got the e2e environment set up properly so I got this merged 👍 you can pull it from develop

Comment (Hyper3x) [reply packages/gateway/src/features/registration/root-resolvers.ts:690]: ok will take the pull, thank you for the update👍

File: packages/client/src/views/DataProvider/birth/mutations.ts [hunk 6]
```diff
@@ -96,6 +96,24 @@ export const MARK_EVENT_AS_DUPLICATE = gql`
...
+  mutation rejectRegistration(
+    $id: String!
+    $details: IdentifierInput!
+  ) {
+    rejectRegistration(
+      id: $id
+      details: $details
+    )
+  }
+`
```
Comment (naftis) [inline packages/client/src/views/DataProvider/birth/mutations.ts:115]: No need to add these @PathumN99

Comment (naftis) [inline packages/client/src/views/DataProvider/birth/mutations.ts:115]: These are unused, no need to add them - @Hyper3x 
I'll remove

**Pull request merged**